File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {

    /**
     * 查询所有
     * @return
     */
    List<TbBrand> findAll();

    /**
     * 分页查询
     * @return
     */
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 新增品牌
     * @param tbBrand
     */
    void add(TbBrand tbBrand);

    /**
     * 批量删除
     * @param ids
     */
    void delete(long[] ids);

    /**
     * 修改品牌信息
     * @param tbBrand
     */
    void update(TbBrand tbBrand);

    /**
     * 根据id查询
     * @param id
     * @return
     */
    TbBrand findOne(long id);

    /**
     * 模糊查询
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(TbBrand tbBrand, Integer pageNum, Integer pageSize);


    /**
     * TypeTemplate商品类型模板编辑选项
     * @return
     */
    List<Map> findOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {
    //查找所有,不分页
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    }
    //查找所有,分页
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../brand/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize);
    }
    //索索,分页
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../brand/search.do?pageNum=' + pageNum + '&pageSize=' + pageSize, searchEntity);
    }

    //添加brand
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    }
    //修改brand
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    }
    //删除brand
    this.del = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    }

    //findOne
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    }

});